h2o-algos/src/main/java/hex/gam/MatrixFrameUtils/GenerateGamMatrixOneColumn.java

```diff
@@ -33,34 +34,51 @@ public GenerateGamMatrixOneColumn(int splineType, int numKnots, double[] knots,
      CubicRegressionSplines crSplines = new CubicRegressionSplines(numKnots, knots);
     _bInvD = crSplines.gen_BIndvD(crSplines._hj);
     _penaltyMat = crSplines.gen_penalty_matrix(crSplines._hj, _bInvD);
-    _gamX = gamx;
+    _maxAbsRowSum = new double[gamx.vec(0).nChunks()];
     _knots = knots;
   }
 
   @Override
   public void map(Chunk[] chk, NewChunk[] newGamCols) {
-    _maxAbsRowSum = new double[_gamX.vec(0).nChunks()];
+    for (int i = 0; i < chk.length; i++) {
+      if (chk[i].len() != chk[0].len()) throw new IllegalArgumentException("Chunks are not the same len"); 
```

Comment:
Ok, this check actually fails. Shouldn't the chunk layout be all the same in a frame @michalkurka @honzasterba and possibly others ? What am I missing ?